decavision/dataset_preparation/generate_pseudolabels.py

```diff
@@ -0,0 +1,216 @@
+import tensorflow as tf
+from tensorflow.keras.models import load_model
+from tqdm import tqdm
+
+
+class PseudoLabelGenerator:
+    """
+    Class to generate pseudo labels.
+    Arguments:
+        model_path (str): path to trained model
+        train_path (str): path to training data
+        unlabeled_path (str): path to unlabeled data
+        pseudo_path (str): path to store train data and pseudo data
+    """
+
+    def __init__(self, model_path, train_data_path, unlabeled_path, pseudo_data_path):
+        self.model_path = model_path
+        self.train_data_path = train_data_path
+        self.unlabeled_path = unlabeled_path
+        self.pseudo_data_path = pseudo_data_path
+
+        # Load model
+        model = load_model(self.model_path, compile=False)
+        self.model = model
+        print("Loaded model.")
+
+    def _load_img(self, path, target_size=(299, 299)):
+        """
+        Load an image from a given path and normalize it
+
+        Args:
+            path (list): Input image path
+            target_size (tuple): Size of image
+        Returns:
+            np.array: Numpy array of the data
+        """
+        # Read image
+        bits = tf.io.read_file(path)
+        image = tf.image.decode_jpeg(bits, channels=3)
+        # Resize
+        image = tf.image.resize(image, size=[*target_size])
+        image = tf.reshape(image, [*target_size, 3])
+        image = tf.cast(image, tf.uint8)
+        # Normalize [0, 1]
+        image = tf.image.convert_image_dtype(image, dtype=tf.float32)
+        image = image.numpy()
+        return image
+
+    def _plot_data(self, predictions, name, output_path):
+        """
+        Plots a bar plot.
+
+        Arguments:
+            predictions (list): List of predictions
+            name (str): Title of the plot
+            output_path (str): Save file using this name
+        """
+        predictions = sorted(predictions)
+        samples = [pred for pred in range(len(predictions))]
+        plt.bar(samples, predictions, color='g')
+        plt.axhline(y=0.5, color='r', linestyle='--')
+        plt.title(name, size=16)
+        plt.xlabel("Number of unlabelled images", size=16)
+        plt.ylim([0.0, 1.0])
+        plt.ylabel("Probability", size=16)
+        plt.tick_params(labelright=True)
+        plt.savefig(output_path, dpi=100)
+        plt.clf()  # clear buffer, otherwise plot overlap!
+
+    def _plot_confidence_scores(self, csv_path, classes):
+        """
+        Save pseudo label names and their confidence scores in a csv file.
+
+        Arguments:
+            csv_path (str): Path to save SCV file
+            classes (list): List containing the class names
+        """
+        dt = pd.read_csv(csv_path)
+        dt['All Class Predictions List'] = dt['All Class Predictions'].apply(
+            lambda x: ast.literal_eval(x))
+
+        raw_predictions_ = dt[["Highest Confidence"]].values
+        raw_predictions = [raw_predictions_[idx][0]
```

Comment:
Maybe this could be done more clearly with the apply method of pandas and a lambda function. In any case, if I understand your code correctly, this list comprehension could be written more simply the following way:
```
[pred[0] for pred in raw_predictions_]

Reply:
Done.